src/NuGet.Services.AzureSearch/SearchService/SearchTextBuilder.cs

```diff
@@ -23,11 +53,81 @@ public string V3Search(V3SearchRequest request)
             return GetLuceneQuery(request.Query);
         }
 
-        private static string GetLuceneQuery(string query)
+        private string GetLuceneQuery(string query)
         {
-            // TODO: query parsing
-            // https://github.com/NuGet/NuGetGallery/issues/6456
-            return query ?? "*";
+            if (string.IsNullOrWhiteSpace(query))
+            {
+                return MatchAllDocumentsQuery;
+            }
+
+            var grouping = _parser.ParseQuery(query.Trim());
+            if (!grouping.Any())
+            {
+                return MatchAllDocumentsQuery;
+            }
+
+            var result = ToAzureSearchQuery(grouping).ToString();
+            if (string.IsNullOrWhiteSpace(result))
+            {
+                return MatchAllDocumentsQuery;
+            }
+
+            return result;
+        }
+
+        private AzureSearchQuery ToAzureSearchQuery(Dictionary<QueryField, HashSet<string>> grouping)
+        {
+            var result = new AzureSearchQuery();
+
+            foreach (var field in grouping)
+            {
+                // Add values that aren't scoped to a field.
+                if (field.Key == QueryField.Any)
+                {
+                    result.AddNonFieldScopedValues(field.Value);
+                }
+                else if (field.Key != QueryField.Invalid)
+                {
+                    // Add values that are scoped to a valid field.
+                    var fieldName = FieldNames[field.Key];
+                    var values = ProcessFieldValues(field.Key, field.Value);
+
+                    result.AddFieldScopedValues(fieldName, values);
+                }
+             }
+
+            return result;
+        }
+
+        private static IEnumerable<string> ProcessFieldValues(QueryField field, IEnumerable<string> values)
+        {
+            switch (field)
+            {
+                // The "packageId" and "owners" fields are case insensitive exact matches
```

Comment:
As mentioned elsewhere I think the analyzers should define the case sensitives --- but really will there be any field at all that is case sensitive?

Reply:
I've reverted my changes for case insensitive exact match semantics, this will re-added by https://github.com/NuGet/NuGetGallery/issues/6920 

(But no, there's no case sensitive fields)